libyul/ObjectParser.cpp

```diff
@@ -198,7 +198,7 @@ void ObjectParser::parseData(Object& _containingObject)
 std::string ObjectParser::parseUniqueName(Object const* _containingObject)
 {
 	expectToken(Token::StringLiteral, false);
-	auto const name = currentLiteral();
+	std::string const name{currentLiteral()};
```

Comment:
Here storing an explicit copy is required as we `advance` before returning from the function.